geojson_pydantic/base.py

```diff
@@ -0,0 +1,68 @@
+"""pydantic BaseModel for GeoJSON objects."""
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, Set
+
+from pydantic import BaseModel, SerializationInfo, field_validator, model_serializer
+
+from geojson_pydantic.types import BBox
+
+
+class _GeoJsonBase(BaseModel):
+    bbox: Optional[BBox] = None
+
+    # These fields will not be included when serializing in json mode
+    # `.model_dump_json()` or `.model_dump(mode="json")`
+    __exclude_if_none__: Set[str] = {"bbox"}
+
+    @property
+    def __geo_interface__(self) -> Dict[str, Any]:
+        """GeoJSON-like protocol for geo-spatial (GIS) vector data.
+
+        ref: https://gist.github.com/sgillies/2217756#__geo_interface
+        """
+        return self.model_dump(mode="json")
```

Comment:
🤯

Reply:
Yeah, this was a nice little fix here when I realized the old `exclude_unset=True` was potentially problematic. Not so much on the Geometry types, but on Feature / FeatureCollection it could affect Properties if someone has defaults set in that model.